server/src/test/java/org/elasticsearch/cluster/coordination/NodeJoinTests.java

```diff
@@ -162,7 +164,18 @@ protected void onSendRequest(long requestId, String action, TransportRequest req
             transportService,
             ESAllocationTestCase.createAllocationService(Settings.EMPTY),
             masterService,
-            () -> new CoordinationStateTests.InMemoryPersistedState(term, initialState), r -> emptyList(), random);
+            () -> new CoordinationStateTests.InMemoryPersistedState(term, initialState), r -> emptyList(),
+            new ClusterApplier() {
```

Comment:
Maybe extract to a `NoOpClusterApplier` rather than adding all these lines here?

Reply:
fixed in a6739a9